src/otx/v2/adapters/torch/mmengine/mmdet/model.py

```diff
@@ -0,0 +1,125 @@
+"""OTX adapters.torch.mmengine.mmpretrain Model APIs."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import fnmatch
+from copy import deepcopy
+from pathlib import Path
+
+import torch
+from mmengine.logging import MMLogger
+from mmengine.registry import MODELS
+from mmengine.runner.checkpoint import load_checkpoint
+
+from otx.v2.adapters.torch.mmengine.modules.utils.config_utils import CustomConfig as Config
+from otx.v2.api.utils.importing import get_files_dict, get_otx_root_path
+from otx.v2.api.utils.logger import LEVEL, get_logger
+
+logger = get_logger()
+
+MODEL_CONFIG_PATH = Path(get_otx_root_path()) / "v2/configs/detection/models"
+MODEL_CONFIGS = get_files_dict(MODEL_CONFIG_PATH)
+
+
+def get_model(
+    model: str | (Config | dict),
+    num_classes: int | None = None,
+    channel_last: bool = False,
+) -> torch.nn.Module:
+    """Return a PyTorch model for training.
+
+    Args:
+        model (Union[str, Config, Dict]): The model to use for training. Can be a string representing the model name,
+            a Config object, or a dictionary.
+        num_classes (Optional[int], optional): The number of classes in the dataset. Defaults to None.
+        channel_last (bool, optional): Whether to use channel last memory format. Defaults to False.
+
+    Returns:
+        torch.nn.Module: The PyTorch model for training.
+    """
+    model_name = None
+    if isinstance(model, dict):
+        model_cfg = Config(cfg_dict=model)
+    elif isinstance(model, str):
+        if Path(model).is_file():
+            model_cfg = Config.fromfile(filename=model)
+        else:
+            model_name = model
+            model_cfg = Config({"model": {}})
+    else:
+        model_cfg = model
+
+    if model_cfg.get("model", None) is None:
+        model_cfg = Config({"model": Config.to_dict(model_cfg)})
+    checkpoint = model_cfg.get("load_from", None)
+    model = Config(model_cfg.get("model"))
+    if hasattr(model, "name"):
+        model_name = model.pop("name")
+
+    if isinstance(model_name, str) and model_name in MODEL_CONFIGS:
+        base_model = Config.fromfile(filename=MODEL_CONFIGS[model_name])
+        checkpoint = base_model.get("load_from", None)
+        base_model = base_model.get("model", base_model)
+        model_cfg.model = Config(cfg_dict=Config.merge_cfg_dict(base_model, model_cfg.model))
+
+    model_cfg["model"]["_scope_"] = "mmdet"
+    if num_classes is not None:
```

Comment:
I think we can use global method to replace num_classes. However, the function you attach will miss some cases. 
Therefore, I think it would be better to create PR for unifying method to replace num_classes after phase 1

Reply:
Okay, let's make global function after phase1